File: common/src/main/java/com/ssblur/scriptor/word/descriptor/discount/HealthyDescriptor.java
package com.ssblur.scriptor.word.descriptor.discount;

import com.ssblur.scriptor.api.word.Descriptor;
import com.ssblur.scriptor.helpers.targetable.EntityTargetable;
import com.ssblur.scriptor.helpers.targetable.Targetable;
import com.ssblur.scriptor.api.word.descriptor.CastDescriptor;
import net.minecraft.world.entity.LivingEntity;

public class HealthyDescriptor extends Descriptor implements CastDescriptor {
  @Override
  public Cost cost() {
    return new Cost(0.9d, COSTTYPE.MULTIPLICATIVE);
  }

  @Override
  public boolean cannotCast(Targetable caster) {
    if(caster instanceof EntityTargetable entityTargetable && entityTargetable.getTargetEntity() instanceof LivingEntity living)
      return living.getHealth() < (living.getMaxHealth() * 0.9);
    return true;
  }
}

File: common/src/main/kotlin/com/ssblur/scriptor/word/descriptor/discount/HealthyDescriptor.kt
package com.ssblur.scriptor.word.descriptor.discount

import com.ssblur.scriptor.api.word.Descriptor
import com.ssblur.scriptor.helpers.targetable.EntityTargetable
import com.ssblur.scriptor.helpers.targetable.Targetable
import com.ssblur.scriptor.word.descriptor.CastDescriptor
import net.minecraft.world.entity.LivingEntity

class HealthyDescriptor : Descriptor(), CastDescriptor {
    override fun cost() = Cost(0.8, COSTTYPE.MULTIPLICATIVE)

    override fun cannotCast(caster: Targetable?): Boolean {
        if (caster is EntityTargetable && caster.targetEntity is LivingEntity) {
            val living = caster.targetEntity as LivingEntity
            return living.getHealth() < living.getMaxHealth()
        }
        return true
    }
}
